client.go

```diff
@@ -132,7 +130,7 @@ func (conn *odpsConn) sign(r *http.Request) {
 			}
 			canonicalResource.WriteString(k)
 			v := urlParams[k]
-			if v != nil && len(v) > 0 && v[0] != "" {
```

Comment:
Why removing the `v != nil` check?

Reply:
`if  len(v) > 0` works same as `if v != nil && len(v) > 0`